assets/scss/components/_button.scss

```diff
@@ -59,6 +59,34 @@
     padding: 0.75rem 1rem;
     margin: 0;
     cursor: pointer;
+
+    &--with-icon {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        gap: rem-calc(8px);
+
+        @include bp(medium) {
+            display: none;
+        }
+
+        &:hover {
+            path {
+                stroke: white;
+                transition: all 0.2s ease-in-out;
+            }
+        }
+
+        svg {
+            transform: rotate(180deg);
+            transition: all 0.2s ease-in-out;
+
+            &.is-open {
```

Comment:
it would make more sense to toggle `is-open` directly on button, instead of the SVG. Then you could do
```
&--icon {
  &.is-open svg { .... }
}
```